examples/dart/closing_labels.lua

```diff
@@ -0,0 +1,9 @@
+-- The configured callback will fetch the labels automatically. You just need
+-- to tell the labels to be drawn.
+vim.cmd [[autocmd DartShowClosingLabels CursorHold,CursorHoldI *.dart :lua require('lsp_extensions.dart.closing_labels').draw_labels()]]
+
+-- With a group
+vim.cmd [[augroup DartShowClosingLabels]]
+vim.cmd [[  au!]]
+vim.cmd [[  autocmd CursorHold,CursorHoldI *.dart :lua require('lsp_extensions.dart.closing_labels').draw_labels()]]
+vim.cmd [[augroup END]]
```

Comment:
We may want to just make these into some functions, so that people can do something like `require('lsp_extensions.dart').setup { close_labels_on_cursor_hold = true }`.

What do you think about that? Then we can just set them up for people.

Reply:
I could definitely include this in the get_callback function so that this autocommand is set up as soon as they opt-in to the callback and make the events customizable in the opts.

I _think_ most people will find this repo after familiarizing themselves with the built-in lsp and the nvim-lspconfig repo. I think defining the api in terms of "how do I plug this into my nvimlsp-config?" will make this easy for people to incorporate into their existing setups. 

From an ease of use perspective, just exposing a `setup` function does make this easy. We could set the callback in `vim.lsp.callbacks`, but I think automatically putting it in the callbacks sort of tramples on the extensibility aspect. 

I'm definitely willing to implement this in more of a `setup` style if that's an api you're interested in exploring but I think there's a lot of value keeping the consumption inline with nvim-lspconfig.